clang/lib/CodeGen/CodeGenFunction.cpp

```diff
@@ -564,6 +564,27 @@ CodeGenFunction::DecodeAddrUsedInPrologue(llvm::Value *F,
                             "decoded_addr");
 }
 
+void CodeGenFunction::EmitSubGroupMetadata(const FunctionDecl *FD,
+                                           llvm::Function *Fn) {
+  if (getLangOpts().OpenCL || getLangOpts().SYCLIsDevice) {
+    if (!FD->hasAttr<OpenCLKernelAttr>())
+      return;
+    CGM.GenOpenCLArgMetadata(Fn, FD, this);
+  }
```

Comment:
I see that `EmitSubGroupMetadata` is called from `EmitOpenCLKernelMetadata` where `GenOpenCLArgMetadata` is called already. It is strange. Why can't we just return to the previous version of `EmitSubGroupMetadata` (where we didn't check on OpenCL, sycldevice and opencl kernel attribute) and call it for only for host from `StartFunction` and if we compile for device/opencl call it also from `EmitOpenCLKernelMetadata`?

Reply:
Thanks @elizabethandrews and @Fznamznon for catching this. I wanted to remove duplicate codes from EmitOpenCLKernelMetadata where we are emitting  "intel_reqd_sub_group_size" for device/opencl call  in my last patch. But i  called GenOpenCLArgMetadata twice. This is fixed now with the current patch.